### PR TITLE
 add primeng table with mock data

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

File: src/app/modules/gb-data-selection-module/table-components/gb-table-dimensions/gb-table-dimensions.component.html [hunk 1]
```diff
@@ -7,8 +7,15 @@
...
               [showTargetControls]="false"
-              [sourceStyle]="{'height':'300px'}"
-              [targetStyle]="{'height':'300px'}">
+              (onMoveToTarget)="updateTable()"
+              (onMoveToSource)="updateTable()"
+              (onMoveAllToTarget)="updateTable()"
+              (onMoveAllToSource)="updateTable()"
+              (onSourceReorder)="updateTable()"
+              (onTargetReorder)="updateTable()"
+              (drop)="onDrop()"
```
Comment (ewelinagr) [inline src/app/modules/gb-data-selection-module/table-components/gb-table-dimensions/gb-table-dimensions.component.html]: wouldn't it be better to add some button to apply changes after setting a desired dimension state? The back-end call is kind of complex, so maybe it's better to limit the number of calls?

File: src/app/modules/gb-data-selection-module/table-components/gb-table-dimensions/gb-table-dimensions.component.ts [hunk 2]
```diff
@@ -9,15 +9,47 @@ import {TableService} from '../../../../services/table.service';
...
+
+  updateTable() {
+    this.tableService.mockDataUpdate();
+  }
+
+  /**
+   * This function handles the drop event when the user is reordering dimensions within
+   * the same row-dimension container or the same column-dimension container
+   */
+  onDrop() {
```
Comment (ewelinagr) [inline src/app/modules/gb-data-selection-module/table-components/gb-table-dimensions/gb-table-dimensions.component.ts]: why is this needed? is the [dragdrop]=true option not working correctly?

File: src/app/modules/gb-data-selection-module/table-components/gb-table-table/gb-table-table.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import {TableService} from '../../../../services/table.service';
+import {Col} from '../../../../models/table-models/col';
+import {Row} from '../../../../models/table-models/row';
+
+@Component({
+  selector: 'gb-table-table',
+  templateUrl: './gb-table-table.component.html',
+  styleUrls: ['./gb-table-table.component.css']
+})
+export class GbTableTableComponent implements OnInit {
```
Comment (ewelinagr) [inline src/app/modules/gb-data-selection-module/table-components/gb-table-table/gb-table-table.component.ts]: I am not sure if it's on purpose, but maybe better to rename it to just GbTableComponent?

Comment (beaugogh) [reply src/app/modules/gb-data-selection-module/table-components/gb-table-table/gb-table-table.component.ts]: yes I will rename it! but to differentiate it from the existing gb-table, I will name it to gb-table-view

File: src/app/modules/gb-data-selection-module/table-components/gb-table-dimensions/gb-table-dimensions.component.ts (hunk 2 above)
Comment (beaugogh) [reply src/app/modules/gb-data-selection-module/table-components/gb-table-dimensions/gb-table-dimensions.component.ts]: primeng does not provide this event, when the user drags to reorder the dimensions, nothing happens! typical

Comment (ewelinagr) [reply src/app/modules/gb-data-selection-module/table-components/gb-table-dimensions/gb-table-dimensions.component.ts]: hmm, the demo on their website seemed working, but then probably they broke something. Ok, so please ignore my comment here
